### PR TITLE
add user with custom UID and GID

### PR DESCRIPTION
**Pull request by jstriebel**
@normanrz: maybe `gosu` is more appropriate, but I'm not sure

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+USER_NAME=${USER_NAME:-sbt}
+
+if [[ -d "/home/${USER_NAME}" ]]; then
+  chown ${USER_UID}:${USER_GID} "/home/${USER_NAME}"
+  # copy user files from /etc/skel
+  install -g ${USER_GID} -o ${USER_UID} /etc/skel/* "/home/${USER_NAME}"
+fi
+
+# create group with USER_GID
+if ! getent group "${USER_NAME}" >/dev/null; then
```
Comment (normanrz) [inline entrypoint.sh]: check for `GID` here

File: entrypoint.sh [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
+  groupadd -f -g ${USER_GID} "${USER_NAME}" 2> /dev/null
+fi
+
+# create user with USER_UID
+if ! getent passwd "${USER_NAME}" >/dev/null; then
```
Comment (normanrz) [inline entrypoint.sh]: check for `UID` here

File: entrypoint.sh [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+  adduser --disabled-login --uid ${USER_UID} --gid ${USER_GID} \
+    --gecos "${USER_NAME}" "${USER_NAME}"
+fi
+
 if [ "$1" = 'sbt' ] || [ "$1" = 'bash' ] || [ "$1" = 'sh' ]; then
-    su "$SBT_USER" -c "exec $@"
+    cd $(eval echo "~${USER_NAME}")
+    exec sudo -HEu "${USER_NAME}" $@
```
Comment (normanrz) [inline entrypoint.sh]: Use `gosu`

File: entrypoint.sh [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
+
+# create user with USER_UID
+if ! getent passwd "${USER_NAME}" >/dev/null; then
+  adduser --disabled-login --uid ${USER_UID} --gid ${USER_GID} \
+    --gecos "${USER_NAME}" "${USER_NAME}"
+fi
+
 if [ "$1" = 'sbt' ] || [ "$1" = 'bash' ] || [ "$1" = 'sh' ]; then
-    su "$SBT_USER" -c "exec $@"
+    cd $(eval echo "~${USER_NAME}")
```
Comment (normanrz) [inline entrypoint.sh]: can `gosu` change to home directory?

File: entrypoint.sh [hunk 5]
```diff
@@ -1,10 +1,32 @@
 #!/usr/bin/env bash
 set -e
 
+USER_UID=${USER_UID:-1000}
+USER_GID=${USER_GID:-$USER_UID}
+USER_NAME=${USER_NAME:-sbt}
+
+if [[ -d "/home/${USER_NAME}" ]]; then
```
Comment (normanrz) [inline entrypoint.sh]: remove this conditional?

File: entrypoint.sh [hunk 6]
```diff
@@ -1,10 +1,32 @@
 #!/usr/bin/env bash
 set -e
```
Comment (normanrz) [inline entrypoint.sh]: `set -Eeuo pipefail`

**Comment by jstriebel** [comment]
done, ping @normanrz

**Review by normanrz — APPROVED** [review]

**Pull request merged**